kythe/go/storage/table/table.go

```diff
@@ -44,6 +44,14 @@ type Proto interface {
 	Close(context.Context) error
 }
 
+// ProtoRO is a read-only key-value direct lookup table with protobuf values.
+// It duplicates the "Lookup" method from the Proto interface, above.
+type ProtoRO interface {
```

Comment:
Rename `ProtoLookup` and embed it within the above `Proto` interface:

```
type Proto interface {
  ProtoLookup

  // Put marshals msg and writes it as the value for the given key.
  Put(ctx context.Context, key []byte, msg proto.Message) error
  ...
}
```

Reply:
TIL that you could have nested interfaces, thanks.  Done.

Reply:
[aside] The effect is analogous to subclassing, in that this requires all the methods of the embedded interface in addition to those named explicitly. (This is related, but slightly different, from embedding in a struct, which grants the struct a field of that type, whose fields and methods are desugared to the enclosing struct).